nano/node/json_handler.cpp

```diff
@@ -283,6 +283,40 @@ nano::amount nano::json_handler::amount_impl ()
 	return result;
 }
 
+nano::amount nano::json_handler::decimal_amount_impl ()
+{
+	nano::amount wholeNumber (0);
+	nano::amount decimalNumber (0);
+	nano::amount result (0);
+	if (!ec)
+	{
+		std::string amount_text (request.get<std::string> ("amount"));
+
+		int dotIndex = amount_text.find (".");
```

Comment:
Let's make sure that there is either zero or one dots. And no other possibility.

Reply:
dotIndex references the position of the dot. It's not the amount of dots. However if the input has multiple dots it will still output an error. Eg 5.3.6 will be converted into 5 as the wholenumber, but the remaining 3.6 can't be converted and the invalid_amount error is returned

Reply:
Can we have a test case for it? We also need a test cases for:
* empty string
* negative zero
* positive (with plus sign)
* number too large
* hex notation (should be rejected)
* scientific notation (should be rejected)
* use of comma character (should be rejected)
* use of comma as fractional part separator (not all countries use dot as separator, should be rejected) 

In general I would favour removing support for negatie numbers to reduce the complexity.